### DISCUSSION TITLE
International shipping method

### DISCUSSION THREAD

**Discussion by Nissrine-laal** [solidusio/solidus#4305] [Troubleshooting] [answered]
Hello
### I just started to maintain a website that was developped by another team.

I'm new to **ROR, spree and solidus,** which r the technologies they used for the website. 

The problem i'm facing now is that i have to create a new functionnality for the website, i need to add an **_international shipping_** method, the shipping fees will be different for **each** country with also a different tax rate. I managed to create the method and display it but i can't seem to wrap my head around how i'm gonna give the admin the possibility to modify each shipping fee in the **dashboard** and how to create a **new table** in database to save the fees and stuff. The actual website calculates the shipping fee **before** displaying the address page in checkout which complicates it a lil bit more.

I'm just curious if anyone have tried before to add the international shipping, how can I do that? What files to add? Controllers? Helpers? Decorators? I'm not even sure what's the difference between all of that..

Any help would be really appreciated. Thanks.

**Comment by jarednorman** [comment]
You do not need to modify the code to support international shipping. Through the admin you can create shipping methods that target each country you ship to. Give each shipping method the correct calculator/pricing and you should be good to go.

**Reply by Nissrine-laal** [reply]
Hey, thanks for your answer. But if i'm trying to ship to all of the countries and i only want one shipping method displayed (International shipping) and for the costs to be calculated after getting the address that the user entered, is it possible ? Without creating a method for each country and different calcucaltors..

**Reply by jarednorman** [reply]
All your shipping methods should have different zones that map to their respective countries so only one would ever show for an order.    You don't need only one shipping method.

**Reply by Nissrine-laal** [reply]
Thanks again for your response, i'm trying out something to make it work since they want everyone to be able to order everywhere. Like even if the person is in UK, they should have the choice to ship to France as an example.

**Comment by kennyadsl** [comment]
Hey @Nissrine-laal did you end up solving this? In case this is solved, can you please mark this as answered?

**Reply by Nissrine-laal** [reply]
Yes i did, thanks

**Comment by Nissrine-laal** [comment, marked as answer]
If anyone having the same issue, what i end up doing was create shipping for eah country, mark it as not available and i didn't link to any zone so it won't show. When the user choose a shipping address, i manually create the shipping method and i call @order_recalculate to have the good adjustements and stuff.
